lib/navigation/preprocess_imu.hpp

```diff
@@ -19,11 +22,32 @@ class ImuPreprocessor {
   SensorChecks checkReliable();
 
   template<std::size_t N>
-  core::Float getSpecificQuartile(const std::array<core::Float, N> &clean_accelerometer_data_copy,
-                                  core::Float quartile);
+  core::Float getSpecificQuartile(const std::array<core::Float, N> &clean_accelerometer_data,
+                                  core::Float quartile)
+  {
+    const core::Float index_quartile       = (num_reliable_accelerometers_ - 1) * quartile;
+    const std::uint8_t index_quartile_high = static_cast<int>(std::ceil(index_quartile));
+    const std::uint8_t index_quartile_low  = static_cast<int>(std::floor(index_quartile));
+    const core::Float quartile_value       = (clean_accelerometer_data.at(index_quartile_high)
+                                        + clean_accelerometer_data.at(index_quartile_low))
+                                       / 2.0;
+    return quartile_value;
+  }
 
   template<std::size_t N>
-  Quartiles getQuartiles(const std::array<core::Float, N> &imu_data);
+  Quartiles getQuartiles(const std::array<core::Float, N> &imu_data)
+  {
+    std::array<core::Float, N> accelerometer_data_copy;
+    std::copy(imu_data.begin(), imu_data.end(), accelerometer_data_copy.begin());
+
+    std::sort(accelerometer_data_copy.begin(), accelerometer_data_copy.end());
+
+    Quartiles quartiles;
+    quartiles.q1     = getSpecificQuartile(accelerometer_data_copy, 0.25);
+    quartiles.median = getSpecificQuartile(accelerometer_data_copy, 0.5);
+    quartiles.q3     = getSpecificQuartile(accelerometer_data_copy, 0.75);
+    return quartiles;
```

Comment:
i'd prefer

```cpp
return {
  .q1 = getSpecificQuartile(...),
  .median = ...,
  ...
};
```

why is this better?
1. we get rid of the mutable variable `quartiles`.
2. this will complain if some fields in `Quartiles` are not properly initialised, thereby catching a source of undefined behaviour.

Reply:
https://pdimov.github.io/blog/2020/09/07/named-parameters-in-c20/